pkg/loki/modules.go

```diff
@@ -480,7 +493,8 @@ func (t *Loki) initQueryFrontend() (_ services.Service, err error) {
 	).Wrap(frontendHandler)
 
 	var defaultHandler http.Handler
-	if t.Cfg.Frontend.TailProxyURL != "" {
+	// If this process also acts as a Querier we don't do any proxying of tail requests
+	if t.Cfg.Frontend.TailProxyURL != "" && !t.ModuleManager.IsModuleRegistered(Querier) {
```

Comment:
This isn't actually testing if querier is in the active dependency tree, but only that it's a registered module, which is always the case

Reply:
I've traditionally used [`isModuleEnabled`](https://github.com/grafana/loki/blob/07ea1ed99c4ef10cbafc4212f589500afa85c59d/pkg/loki/loki.go#L196) which I agree is a bit hacky, but it checks for the string in the target. I think if we looked for the `Querier`, `All`, and `Read` targets we should be covered.

Reply:
Great catch, 😠 that I missed this!

Reply:
Yeah i went this approach because the `isModuleEnabled` required supplying multiple targets and I thought this approach would work better...

Instead I've added a method which does exactly what I want

Reply:
do we still need both versions of that function?